pkg/hoist/hoist_launchable.go

```diff
@@ -34,25 +35,28 @@ type EntryPoints struct {
-	CgroupConfig     cgroups.Config             // Cgroup parameters to use with p2-exec
-	CgroupConfigName string                     // The string in PLATFORM_CONFIG to pass to p2-exec
-	CgroupName       string                     // The name of the cgroup to run this launchable in
-	RequireFile      string                     // Do not run this launchable until this file exists
-	RestartTimeout   time.Duration              // How long to wait when restarting the services in this launchable.
-	RestartPolicy_   runit.RestartPolicy        // Dictates whether the launchable should be automatically restarted upon exit.
-	SuppliedEnvVars  map[string]string          // A map of user-supplied environment variables to be exported for this launchable
-	Location         *url.URL                   // URL to download the artifact from
-	VerificationData auth.VerificationData      // Paths to files used to verify the artifact
-	EntryPoints      EntryPoints                // paths to entry points to launch under runit
+	Id                launch.LaunchableID        // A (pod-wise) unique identifier for this launchable, used to distinguish it from other launchables in the pod
+	Version           launch.LaunchableVersionID // A version identifier
+	PodID             types.PodID                // A (possibly-null) PodID denoting which launchable this belongs to
+	ServiceId         string                     // A (host-wise) unique identifier for this launchable, used when creating runit services
+	RunAs             string                     // The user to assume when launching the executable
+	OwnAs             string                     // The user that owns all the launcable's artifacts
+	PodEnvDir         string                     // The value for chpst -e. See http://smarden.org/runit/chpst.8.html
+	RootDir           string                     // The root directory of the launchable, containing N:N>=1 installs.
+	P2Exec            string                     // Struct that can be used to build a p2-exec invocation with appropriate flags
+	ExecNoLimit       bool                       // If set, execute with the -n (--no-limit) argument to p2-exec
+	PodCgroupConfig   cgroups.Config             // PodCgroupConfig
+	ResourceLimitPath string                     // ResourceLimitPath points to the resource limits config file
+	CgroupConfig      cgroups.Config             // Cgroup parameters to use with p2-exec
+	CgroupConfigName  string                     // The string in PLATFORM_CONFIG to pass to p2-exec
+	CgroupName        string                     // The name of the cgroup to run this launchable in
+	RequireFile       string                     // Do not run this launchable until this file exists
+	RestartTimeout    time.Duration              // How long to wait when restarting the services in this launchable.
+	RestartPolicy_    runit.RestartPolicy        // Dictates whether the launchable should be automatically restarted upon exit.
```

Comment:
Why does this have an underscore?

Reply:
This isn't my code, but I believe this avoids the golang gotcha that a field name cannot match a function name.